### PR TITLE
Add checksums

### PR DESCRIPTION
**Pull request by mbollmann**
This PR adds and enforces checksums in the XML files, following the discussion in #295.

It makes some substantial changes:

1. Remove pointers to files that are missing on the server (435ef75) and add several that were previously not linked to in the XML (498ea1c).
2. Add a `<revision id="1" />` entry to denote the original version of a paper whenever there exists a revision (5aa0f7b); this is so we can actually store a checksum for the v1 of a paper.
3. Add checksums for all files (5e4140b) and require them in the RelaxNG schema (aa5174d). I've tested that the latter works as expected, though unfortunately, the error message in case of a missing hash is quite unhelpful:

   `error: character content of element "url" invalid; must be a URI matching the regular expression "https?://.*"`

   So apparently the validator assumes that if the `hash` is missing, we must have meant an external URL. Changing the order of expressions didn't change this behavior, so I guess this is just something we have to keep in mind.
4. Add scripts to verify checksums of files or of a filelist generated on the server (85aeca0).
5. Modify the scripts that add attachments & revisions to compute and add the required checksums (28b6aa2). I have *not* made any modifications to the ingestion script, also because I'm not sure which ingestion script is supposed to be used moving forward, but hopefully with these other scripts as a template, it should be easy for @mjpost to make the required changes.

### CODE REVIEW

**Comment by mbollmann** [comment]
Oh, one thing to add could be displaying the checksums on the website itself. I didn't want to postpone the PR for this, though.

**Comment by mjpost** [comment]
This looks good to me. It's nice that you updated `add_revision.py` and `add_attachment.py`, though normally I would have just done this the first time I needed them.

`ingest.py` is the ingestion script. The new `ingest_tsv.py` is for merging externally-source data (IWPT, MT Archive) that has been digitized into TSV files. But yes, I will just update this the next time I do an ingest (which will probably be LREC, which is looming).

I looked over this PR but haven't done any testing. I think we should merge this ASAP so as to reduce the risk of other PRs diverging from this and complicating it.

Maybe one other person should have a glance + think before we merge?

**Comment by mjpost** [comment]
(@mbollmann, I'll leave the squashing + merging to you. I'll hold off on #811 until this is done).

**Comment by akoehn** [comment]
I can have a look at it today if you want to wait for that.

File: data/xml/schema.rnc [hunk 1]
```diff
@@ -11,26 +11,36 @@ first = element first { text }
 last = element last { text }
 Person = attribute id { xsd:NCName }?, (first? & last)
 
+checksum = attribute hash { xsd:string {minLength="8" maxLength="8" pattern="[0-9a-f]*"} }
+url-with-checksum = element url {
+    xsd:anyURI {pattern="https?://.*"} |
+    (xsd:anyURI {pattern=".*"}, checksum)
```
Comment (akoehn) [inline data/xml/schema.rnc:17]: Why did you add the .* pattern here? If it really does something useful, maybe a short comment would be useful.  Restricting by .* seems to be contradictory otherwise.

File: data/xml/schema.rnc [hunk 2]
```diff
@@ -11,26 +11,36 @@ first = element first { text }
...
+}
+attachment = element attachment {
+    attribute type { xsd:NCName }?,
+    checksum,
+    xsd:anyURI
```
Comment (akoehn) [inline data/xml/schema.rnc:22]: This as NCName before, i.e. a non-qualified URI.  Now it can be anything. I don't think it makes sense to allow absolute URIs for attachments.

File: data/xml/schema.rnc (hunk 1 above)
Comment (mbollmann) [reply data/xml/schema.rnc:17]: Ah, I started from your suggestion which included `pattern="/.*"`, but the slash didn't make sense so I removed it and left it like that without thinking further. You're right that it's superfluous.

File: data/xml/schema.rnc (hunk 2 above)
Comment (mbollmann) [reply data/xml/schema.rnc:22]: NCName is not allowed to start with a digit; files named after new-style IDs will start with digits.

Comment (mbollmann) [reply data/xml/schema.rnc:22]: We could probably define an `xsd:string` with a hand-crafted pattern and name it `local-file` or something?

File: data/xml/schema.rnc (hunk 1 above)
Comment (akoehn) [reply data/xml/schema.rnc:17]: I would go back to "/.*" to denote that it can only be local.  If that was not obvious beforehand, maye add a comment à la:
```
    // We do not control the content of external URLS, therefore they have no checksums
    xsd:anyURI {pattern="https?://.*"} |
    // All data hosted by us (with relative paths, i.e. starting with /) need to have a checksum
    (xsd:anyURI {pattern="/.*"}, checksum)
```

File: data/xml/schema.rnc (hunk 2 above)
Comment (akoehn) [reply data/xml/schema.rnc:22]: That would probably the best way, so scratch my recommendation re /.* I made above :-)

Comment (mbollmann) [reply data/xml/schema.rnc:22]: Okay, I'll see if I can make that work. Our relative path entries don't start with a slash in the XML, so `/.*` wouldn't currently match anything, which is why your other suggestion confused me. :)

**Comment by mbollmann** [comment]
Thanks! Do we have a preference for squash merges? I'd pick rebase-and-merge for this one otherwise since it has several changes that stand on their own.

**Comment by akoehn** [comment]
It is your code and in the grand scheme of things there is probably no big difference.  A perfectionist could clean up everything using `git rebase -i` but a simple git rebase should to in this case I think.

**Comment by mjpost** [comment]
I typically do squash-and-merge, but splitting into a few meaningful squashes is fine. In general I like semantic separation of merges, but I want to avoid all the nit-picky intermediate merges (e.g., "black" :)).

**Comment by mbollmann** [comment]
Just for curiosity: Can you `git rebase -i` and then force-push to the branch to rewrite its history on Github? I didn't want to be that perfectionist this time since I thought the commit history was reasonably clean, but it'd be good to know. :)

**Comment by mjpost** [comment]
We've disabled force pushing to `master`, obviously, but I'm not sure about other branches. Try it!
